src/java/org/jivesoftware/openfire/group/GroupManager.java

```diff
@@ -713,4 +765,15 @@ private void evictCachedPaginatedGroupNames() {
             }
         }
     }
+
+    private void evictCachedUserSharedGroups() {
+        synchronized (USER_SHARED_GROUPS_KEY.intern()) {
+            for(Map.Entry<String, Object> entry : groupMetaCache.entrySet())
+            {
+                if (entry.getKey().startsWith(USER_SHARED_GROUPS_KEY)) {
+                    groupMetaCache.remove(entry.getKey());
```

Comment:
It's probably unwise to remove items from the collection while iterating over it.